Sources/Public/DotLottie/DotLottieFileHelpers.swift

```diff
@@ -0,0 +1,204 @@
+  public static func asset(
+    _ name: String,
+    bundle: Bundle = Bundle.main,
+    dotLottieCache: DotLottieCacheProvider? = DotLottieCache.sharedCache)
+    -> DotLottieFile?
+  {
+    /// Create a cache key for the lottie.
+    let cacheKey = bundle.bundlePath + "/" + name
+
+    /// Check cache for lottie
+    if
+      let dotLottieCache = dotLottieCache,
+      let lottie = dotLottieCache.file(forKey: cacheKey)
+    {
+      /// If found, return the lottie.
+      return lottie
+    }
+
+    /// Load data from Asset
+    guard let data = Data.jsonData(from: name, in: bundle) else {
+      return nil
+    }
+
+    do {
+      /// Decode lottie.
+      let lottie = try DotLottieFile.from(data: data, filename: name)
+      dotLottieCache?.setFile(lottie, forKey: cacheKey)
+      return lottie
+    } catch {
+      /// Decoding error.
+      return nil
+    }
+  }
+
+  /// Loads a DotLottie animation from a `Data` object containing a compressed .lottie file.
+  ///
+  /// - Parameter data: The object to load the file from.
+  /// - Parameter filename: The name of the file.
+  /// - Returns: Deserialized `DotLottie`. Optional.
+  ///
+  public static func from(data: Data, filename: String) throws
+    -> DotLottieFile
+  {
+    try DotLottieFile(data: data, filename: filename)
+  }
+
+  /// Loads a DotLottie animation asynchronously from the URL.
+  ///
+  /// - Parameter url: The url to load the animation from.
+  /// - Parameter closure: A closure to be called when the animation has loaded.
+  /// - Parameter animationCache: A cache for holding loaded animations. Defaults to `LRUAnimationCache.sharedCache`. Optional.
+  ///
+  public static func loadedFrom(
+    url: URL,
+    session: URLSession = .shared,
+    closure: @escaping DotLottieFile.DotLottieDownloadClosure,
+    dotLottieCache: DotLottieCacheProvider? = DotLottieCache.sharedCache)
+  {
+    if let dotLottieCache = dotLottieCache, let animation = dotLottieCache.file(forKey: url.absoluteString) {
+      closure(animation)
+    } else {
+      let task = session.dataTask(with: url) { data, _, error in
+        guard error == nil, let data = data else {
+          DispatchQueue.main.async {
+            closure(nil)
+          }
+          return
+        }
+        do {
+          let lottie = try DotLottieFile.from(data: data, filename: url.deletingPathExtension().lastPathComponent)
+          DispatchQueue.main.async {
+            dotLottieCache?.setFile(lottie, forKey: url.absoluteString)
+            closure(lottie)
+          }
+        } catch {
+          DispatchQueue.main.async {
+            closure(nil)
+          }
+        }
+      }
+      task.resume()
+    }
+  }
+
+  /// Returns animation with id
+  /// - Parameter id: id to animation. Specified in .lottie file manifest. Optional
+  /// Defaults to first animation
+  /// - Returns: LottieAnimation with id. Optional
+  public func animation(id: String? = nil) -> LottieAnimation? {
```

Comment:
I think this would be more idiomatic

```suggestion
  public func animation(for id: String? = nil) -> LottieAnimation? {
```